core/src/main/java/org/wildfly/channel/ChannelImpl.java

```diff
@@ -202,6 +202,11 @@ Optional<ResolveLatestVersionResult> resolveLatestVersion(String groupId, String
         requireNonNull(artifactId);
         requireNonNull(resolver);
 
+        Set<String> blocklistedVersions = null;
```

Comment:
a small nitpick - if the `blocklistedVersions` was set to `Collections.emptySet()` instead of `null`, we could remove all of the `!= null` checks later on

Reply:
I used the null value as indication that the blocklist is not present and avoid additional operations. It's true that using emptySet() makes the code cleaner, but more calls will be made so I changed the tests to use mutable sets for the artifact versions because the new implementation calls removeAll() on the sets which causes UnsupportedOperationException.